octeon: add support for Linux 6.12 and remove 6.6

rsalvaterra opened the pull request:
Run-tested on a Ubiquiti UniFi Security Gateway (the only Octeon target device I have), built with GCC 15.2 and Binutils 2.45, without any observable issues. Testing on other devices is most welcome.

Signed-off-by: Rui Salvaterra \<rsalvaterra@gmail.com\>

~NOTE: The GCC 15 build fix kernel patch should be temporary. I haven't officially sent a patch upstream, but I will soon if upstream doesn't beat me to the fix.~

rsalvaterra committed:
octeon: refresh 6.12 patches

All but one needs (automatic) rebasing.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra committed:
octeon: update Linux 6.12 kconfig

Update with make kernel_oldconfig CONFIG_TARGET=target. Also disable
CONFIG_POSIX_MQUEUE, as it's available in the OpenWrt configuration.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra committed:
octeon: set kernel 6.12 as default and remove support for 6.6

Get the Octeon target ready for the next OpenWrt release.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra committed:
octeon: fix command line hack for Linux 6.12

Since strlcpy has been removed in Linux 6.8, replace it with strscpy.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

andrewjlamarche commented:
I have a currently unsupported board based on evb7000_sff that I've ported (most) support for. I'll give this a test sometime soon, though it may not hold merit since it is not an officially supported device.

robimarko commented:
@rsalvaterra https://lists-ec2.96boards.org/archives/list/linux-stable-mirror@lists.linaro.org/thread/VT3IZRPFK6IAHNF42246LU76O6CEGGKS/

It has landed in linux-next, so you can just backport it:
https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/arch/mips/vdso?h=next-20250502&id=0f4ae7c6ecb89bfda026d210dcf8216fb67d2333

rsalvaterra commented:
> @rsalvaterra https://lists-ec2.96boards.org/archives/list/linux-stable-mirror@lists.linaro.org/thread/VT3IZRPFK6IAHNF42246LU76O6CEGGKS/
> 
> It has landed in linux-next, so you can just backport it: https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/arch/mips/vdso?h=next-20250502&id=0f4ae7c6ecb89bfda026d210dcf8216fb67d2333

Nice, thanks for the heads up, @robimarko. In that case, I rather just wait for the next stable kernel release containing the aforementioned fix. There's no need for extra churn.

robimarko commented:
Works for me :)

andrewjlamarche commented:
Looks like this is missing https://github.com/openwrt/openwrt/blob/main/target/linux/octeon/patches-6.6/702-qca833x-force-pcs-reset.patch

robimarko commented:
@rsalvaterra https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?h=next-20250606&id=0f4ae7c6ecb89bfda026d210dcf8216fb67d2333

Since stable is clearly not picking it up please use it instead of your patch

rsalvaterra commented:
> @rsalvaterra https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?h=next-20250606&id=0f4ae7c6ecb89bfda026d210dcf8216fb67d2333
> 
> Since stable is clearly not picking it up please use it instead of your patch

The fix has been merged on 6.12.35, the current version in main. Rebased an force pushed, let's see if CI doesn't complain.

sur5r commented:
Could you pull in `400-ubnt_dts_pruning.patch` as well? I can give it a try on e200 afterwards.

rsalvaterra commented:
> Could you pull in `400-ubnt_dts_pruning.patch` as well? I can give it a try on e200 afterwards.

Done, thanks for the heads up!

andrewjlamarche commented:
Looks like the kernel is still failing to build.

rsalvaterra commented:
I see the problem. I'm building locally with `CONFIG_KERNEL_WERROR` disabled. And there are issues with the kernel proper, not only our patches. I guess upstream hasn't built Octeon kernels in a long time…

sur5r commented:
Ah, I see where I went wrong. I didn't turn on `CONFIG_TESTING_KERNEL`.

rsalvaterra committed:
octeon: fix octeon build with CONFIG_KERNEL_WERROR

Add a patch fixing a bunch of missing prototype errors, scattered all over the
tree.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra commented:
Updated. Hopefully this time CI will be happy.

andrewjlamarche commented:
Will test soon, though on a not-officially-supported device. Don’t want the octeon target to be dropped!

andrewjlamarche commented:
```
root@OpenWrt:~# ubus call system board
{
	"kernel": "6.12.40",
	"hostname": "OpenWrt",
	"system": "Octeon EVB7000_SFF (CN7030p1.2-1000-AAP)",
	"model": "Octeon EVB7000_SFF",
	"board_name": "octeon,evb7000_sff",
	"rootfs_type": "squashfs",
	"release": {
		"distribution": "OpenWrt",
		"version": "SNAPSHOT",
		"firmware_url": "https://downloads.openwrt.org/",
		"revision": "r30731+6-886c4f054e",
		"target": "octeon/generic",
		"description": "OpenWrt SNAPSHOT r30731+6-886c4f054e",
		"builddate": "1754772706"
	}
}
```

Booted on the (unsupported) dev board, though I can't seem to get LAN ports 2-4 to work. I don't touch this board often, perhaps it was still an issue on 6.6. I'll have to revisit 6.6.

Maybe @Grommish can test on an officially supported board.

Djfe commented:
please recreate some of these commits using the kernel bump script to keep the commit history
https://github.com/openwrt/openwrt/pull/19136#issuecomment-2989117473

rsalvaterra committed:
kernel/octeon: Create kernel files for v6.12 (from v6.6)

This is an automatically generated commit.

When doing `git bisect`, consider `git bisect --skip`.

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra committed:
kernel/octeon: Restore kernel files for v6.6

This is an automatically generated commit which aids following Kernel patch
history, as git will see the move and copy as a rename thus defeating the
purpose.

For the original discussion see:
https://lists.openwrt.org/pipermail/openwrt-devel/2023-October/041673.html

Signed-off-by: Rui Salvaterra <rsalvaterra@gmail.com>

rsalvaterra commented:
@namiltd, can you please update #16569 accordingly?

rsalvaterra commented:
Rebased and merged.
